File: src/backend/src/main/java/br/com/grupopipa/gestaointegrada/atendimento/agendamento/agenda/dto/AgendaGridDTO.java
package br.com.grupopipa.gestaointegrada.atendimento.agendamento.agenda.dto;

import java.time.LocalDateTime;
import java.util.UUID;

import br.com.grupopipa.gestaointegrada.core.dto.DTO;
import br.com.grupopipa.gestaointegrada.core.dto.GridDTO;
import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class AgendaGridDTO implements DTO, GridDTO {
    private UUID id;
    private String nome;
    private String profissionalNome;
    private String setorNome;
    private Boolean ativo;
    private LocalDateTime createdAt;
    private Boolean deleted;
}

File: src/frontend/src/app/components/atendimento/agendamento/agenda/model/agenda-grid-dto.ts
import { Exclude, Expose } from 'class-transformer';

@Exclude()
export class AgendaGridDTO {
  @Expose() id?: string;
  @Expose() nome?: string;
  @Expose() profissionalNome?: string;
  @Expose() setorNome?: string;
  @Expose() ativo?: boolean;
  @Expose() createdAt?: string;
  @Expose() deleted?: boolean;
}
